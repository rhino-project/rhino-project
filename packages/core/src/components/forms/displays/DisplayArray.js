import { useCallback } from 'react';
import FieldInputControlled from '../fields/FieldInputControlled';

const DisplayArray = ({ accessor: propAccessor, empty = '-', ...props }) => {
  const accessor = useCallback(
    (value) => {
      // If the accessor prop is defined, use it to get the value to display.
      const newValue = propAccessor ? propAccessor(value) : value;

      if (newValue == null) return empty;

      return newValue;
    },
    [empty, propAccessor]
  );

  return (
    <FieldInputControlled type="text" accessor={accessor} readOnly {...props} />
  );
};

export default DisplayArray;
