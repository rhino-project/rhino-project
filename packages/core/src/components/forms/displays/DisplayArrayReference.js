import { useCallback } from 'react';
import { useGlobalComponent } from '../../../hooks/overrides';
import DisplayString from './DisplayString';

export const DisplayArrayReferenceBase = (props) => {
  const accessor = useCallback(
    (value) => value?.map((v) => v.display_name)?.join(', '),
    []
  );

  return <DisplayString {...props} accessor={accessor} />;
};

const DisplayArrayReference = (props) =>
  useGlobalComponent('DisplayArrayReference', DisplayArrayReferenceBase, props);

export default DisplayArrayReference;
