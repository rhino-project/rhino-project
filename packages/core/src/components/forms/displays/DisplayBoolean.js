import { useGlobalComponent } from '../../../hooks/overrides';
import FieldBooleanIndeterminate from '../fields/FieldBooleanIndeterminate';

export const DisplayBooleanBase = (props) => (
  <FieldBooleanIndeterminate readOnly {...props} />
);

export const DisplayBoolean = (props) =>
  useGlobalComponent('DisplayBoolean', DisplayBooleanBase, props);
