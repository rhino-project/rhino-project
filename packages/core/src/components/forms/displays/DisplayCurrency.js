import { useGlobalComponent } from '../../../hooks/overrides';
import { InputGroup } from 'reactstrap';
import { applyCurrencyMask } from '../../../utils/ui';
import DisplayString from './DisplayString';

export const DisplayCurrencyBaseInput = (props) => {
  return (
    <DisplayString
      type="text"
      accessor={applyCurrencyMask}
      readOnly
      {...props}
    />
  );
};

export const DisplayCurrencyBase = (props) => {
  return (
    <InputGroup>
      <span className="input-group-text">$</span>
      <DisplayCurrencyBaseInput {...props} />
    </InputGroup>
  );
};

export const DisplayCurrency = (props) =>
  useGlobalComponent('DisplayCurrency', DisplayCurrencyBase, props);
