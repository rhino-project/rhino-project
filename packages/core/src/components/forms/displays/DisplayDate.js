import { useCallback } from 'react';
import { format as dateFormat, parseISO } from 'date-fns';
import { FieldInputControlledBase } from '../fields/FieldInputControlled';
import { useGlobalComponent } from 'rhino/hooks/overrides';

export const DisplayDateBase = ({
  format = 'MMMM d, yyyy',
  empty = '-',
  ...props
}) => {
  const accessor = useCallback(
    (value) => {
      if (!value) return empty;

      return dateFormat(parseISO(value), format);
    },
    [empty, format]
  );

  return (
    <FieldInputControlledBase
      type="text"
      accessor={accessor}
      readOnly
      {...props}
    />
  );
};

const DisplayDate = (props) =>
  useGlobalComponent('DisplayDate', DisplayDateBase, props);

export default DisplayDate;
