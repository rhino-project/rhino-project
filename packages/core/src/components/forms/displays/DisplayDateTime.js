import { useCallback } from 'react';
import { format as dateFormat, parseISO } from 'date-fns';
import { FieldInputControlledBase } from '../fields/FieldInputControlled';
import { useGlobalComponent } from 'rhino/hooks/overrides';

export const DisplayDateTimeBase = ({
  format = 'MMMM d, yyyy h:mm aa',
  empty = '-',
  ...props
}) => {
  const accessor = useCallback(
    (value) => {
      if (!value) return empty;

      return dateFormat(parseISO(value), format);
    },
    [empty, format]
  );

  return (
    <FieldInputControlledBase
      type="text"
      accessor={accessor}
      readOnly
      {...props}
    />
  );
};

const DisplayDateTime = (props) =>
  useGlobalComponent('DisplayDateTime', DisplayDateTimeBase, props);

export default DisplayDateTime;
