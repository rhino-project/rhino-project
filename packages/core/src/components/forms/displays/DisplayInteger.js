import { useCallback } from 'react';
import { FieldInputControlledBase } from '../fields/FieldInputControlled';
import { useGlobalComponent } from '../../../hooks/overrides';

export const DisplayIntegerBase = ({ empty = '-', ...props }) => {
  const accessor = useCallback(
    (value) => {
      if (value == null) return empty;

      return value;
    },
    [empty]
  );

  return (
    <FieldInputControlledBase
      type="text"
      accessor={accessor}
      readOnly
      {...props}
    />
  );
};

const DisplayInteger = (props) =>
  useGlobalComponent('DisplayInteger', DisplayIntegerBase, props);

export default DisplayInteger;
