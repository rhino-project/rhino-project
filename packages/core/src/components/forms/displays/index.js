export * from './DisplayArray';
export * from './DisplayArrayReference';
export * from './DisplayBoolean';
export * from './DisplayCurrency';
export * from './DisplayDate';
export * from './DisplayDateTime';
export * from './DisplayEnum';
export * from './DisplayFloat';
export * from './DisplayImage';
export * from './DisplayInteger';
export * from './DisplayLink';
export * from './DisplayReference';
export * from './DisplayString';
export * from './DisplayStringControlled';
export * from './DisplayText';
export * from './DisplayTime';
