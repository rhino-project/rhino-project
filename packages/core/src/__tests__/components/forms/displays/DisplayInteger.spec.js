import { sharedDisplayTests } from './sharedDisplayTests';
import { DisplayInteger } from '../../../../components/forms/displays/DisplayInteger';

describe('DisplayInteger', () => {
  sharedDisplayTests(DisplayInteger);
});
