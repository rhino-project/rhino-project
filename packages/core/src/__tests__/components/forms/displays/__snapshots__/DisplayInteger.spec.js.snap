// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DisplayInteger > renders inside of model context 1`] = `
<DocumentFragment>
  <input
    aria-invalid="false"
    autocomplete="off"
    class="form-control"
    id="dummy"
    name="dummy"
    path="dummy"
    readonly=""
    type="text"
    value="-"
  />
</DocumentFragment>
`;

exports[`DisplayInteger > should render with global override shorthand 1`] = `
<DocumentFragment>
  <div>
    Bar
  </div>
</DocumentFragment>
`;
